consensus/safety-rules/src/persistent_safety_storage.rs

```diff
@@ -61,6 +79,16 @@ impl PersistentSafetyStorage {
             .map_err(|e| e.into())
     }
 
+    pub fn execution_public_key(&self) -> Result<Ed25519PublicKey> {
+        Ok(if let Some(key) = self.execution_public_key.as_ref() {
+            key.clone()
+        } else {
+            self.internal_store
+                .get_public_key(EXECUTION_KEY)
+                .map(|r| r.public_key)?
+        })
```

Comment:
this is some really unhappy logic to deal with testing.

Reply:
test_config has an execution keypair and privkey can only be taken once. If I take out private key here, then LEC will get `None` and throw error. Another way is to copy the privkey somehow but I feel it is more hacky.